### PR TITLE
Add baseline grid conditionally.

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Introduces a namespace (vf) and an environment variable vf.ENV. If vf.ENV === "DEV", proceeds to create a baseline grid with just as many baselines as needed to cover the document height. Updates grid on resize. 

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- [Add additional steps]

## Details

[List of links to issues/bugs and cards if needed]

## Screenshots

[if relevant, include a screenshot]

### CODE REVIEW

File: _jekyll/_layouts/default.html [hunk 1]
```diff
@@ -13,6 +13,92 @@
...
+                    this.target.appendChild(fragment);
+                }
+                initGridToggle(switchBtn) {
+                    var toggle = document.querySelector(switchBtn),
+                        hideClass = this.toggleGridClass,
+                        that = this;
+
+                    toggle.addEventListener('click', function (event) {
+                        var grid = document.querySelector('.' + that.baselineContainerClassName);
+                        if(!that.showGrid) {
```
Comment (anthonydillon) [inline _jekyll/_layouts/default.html]: space please

File: _jekyll/_layouts/default.html [hunk 2]
```diff
@@ -13,6 +13,92 @@
...
+                            this.setAttribute('aria-checked', !this.getAttribute('aria-checked'));
```
Comment (anthonydillon) [inline _jekyll/_layouts/default.html]: This should set aria-checked to true.

File: _jekyll/_layouts/default.html [hunk 3]
```diff
@@ -13,6 +13,92 @@
...
+                var body = document.body,
+                    html = document.documentElement;
+                    return Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
+            }
+            vf.fragmentFromString = function fragmentFromString(strHTML) {
+                var temp = document.createElement('template');
+                temp.innerHTML = strHTML;
+                return temp.content;
+            }
+            vf.baselineGrid = class {
```
[the remark this answers is not in the document]
Comment (lyubomir-popov) [reply _jekyll/_layouts/default.html]: probably overkill, just didn't want to pollute the global scope and thought I'd namespace anything I add similar to how the css is namespaced. will remove it.

File: _jekyll/_layouts/default.html [hunk 4]
```diff
@@ -13,6 +13,92 @@
     </head>
 
     <body style="margin: 1rem;">
+
 {{ content }}
     </body>
+    <script>
+        var vf = vf || {};
+        vf.ENV = "DEV";
+        if(vf.ENV === "DEV") {
```
[the remark this answers is not in the document]
Comment (lyubomir-popov) [reply _jekyll/_layouts/default.html]: the idea was to be able to turn it on in development and off in production without having to delete or comment out anything. So my plan was to have this as an environment variable that turns off this and potentially other debug features.

File: _jekyll/_layouts/default.html [hunk 5]
```diff
@@ -13,6 +13,92 @@
...
 {{ content }}
     </body>
+    <script>
+        var vf = vf || {};
+        vf.ENV = "DEV";
+        if(vf.ENV === "DEV") {
+            vf.getDocumentHeight = function getDocumentHeight() {
+                var body = document.body,
+                    html = document.documentElement;
+                    return Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
```
[the remark this answers is not in the document]
Comment (lyubomir-popov) [reply _jekyll/_layouts/default.html]: is it ok to replace with document.documentElement.scrollHeight as per the answer with 81 votes here https://stackoverflow.com/questions/1145850/how-to-get-height-of-entire-document-with-javascript

**Comment by lyubomir-popov** [comment]
@anthonydillon could you please review again? I think I've fixed all the issues you mentioned and would be good to have the grid in while working on other components.

**Comment by anthonydillon** [comment]
@lyubomir-popov thanks, works well now. When we go live with this version (1.7.0) I think we'll have to turn the grid off by default. Otherwise all the examples in the docs will have the grid on without switching it.

**Comment by lyubomir-popov** [comment]
@anthonydillon My plan was to set VF_ENV to "PRODUCTION" or something like that when this goes live, and the whole script won't run.

Added 2 final commits - 
1) turn the grid off on page load
2) revert line height to avoid percy differences - as it should be on a separate branch anyway.

**Comment by anthonydillon** [comment]
@lyubomir-popov I still see the toggle on by default.

**Comment by anthonydillon** [comment]
Sorry, what I mean is that the demo still shows the toggle as active but the grid is hidden.

**Comment by anthonydillon** [comment]
Thanks :+1:

**Pull request merged**